File: Java/Medium/2559.java
class Solution {
    public int[] vowelStrings(String[] words, int[][] queries) {
        // hold the prefix sum
        int[] sumArr = new int[words.length];
        for (int i = 0; i < words.length; i++) {
            if (words[i].charAt(0) == 'a' || words[i].charAt(0) == 'e' || words[i].charAt(0) == 'i'
                    || words[i].charAt(0) == 'o' || words[i].charAt(0) == 'u') {
                if (words[i].charAt(words[i].length() - 1) == 'a'
                        || words[i].charAt(words[i].length() - 1) == 'e'
                        || words[i].charAt(words[i].length() - 1) == 'i'
                        || words[i].charAt(words[i].length() - 1) == 'o'
                        || words[i].charAt(words[i].length() - 1) == 'u') {
                    sumArr[i] = (i == 0) ? 1 : sumArr[i - 1] + 1;
                } else {
                    sumArr[i] = (i == 0) ? 0 : sumArr[i - 1];
                }
            } else {
                sumArr[i] = (i == 0) ? 0 : sumArr[i - 1];
            }
        }

        int[] answer = new int[queries.length];

        for (int j = 0; j < queries.length; j++) {
            answer[j] = (queries[j][0] == 0) ? sumArr[queries[j][1]]
                    : sumArr[queries[j][1]] - sumArr[queries[j][0] - 1];
        }

        return answer;
    }
}

File: Python/Medium/2559.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        # use a prefix sum method
        sumArr = list()

        # use a set to store the vowels
        vowels = {"a", "e", "i", "o", "u"}

        for word in words:
            if (word[0] in vowels) and (word[-1] in vowels):
                if len(sumArr) == 0:
                    sumArr.append(1)
                else:
                    sumArr.append(sumArr[-1] + 1)
            else:
                if len(sumArr) == 0:
                    sumArr.append(0)
                else:
                    sumArr.append(sumArr[-1])

        answer = list()

        for query in queries:
            if query[0] == 0:
                answer.append(sumArr[query[1]])
            else:
                answer.append(sumArr[query[1]] - sumArr[query[0] - 1])

        return answer
